pkg/status/conditions.go

```diff
@@ -0,0 +1,241 @@
+		ac.LastTransitionTime = metav1.Now()
+		return ac
+	}
+	// MonitoringStack status will not be updated if there is a difference between the operand generation
+	// and the operand ObservedGeneration. This can occur, for example, in the case of an invalid operand configuration.
+	if operandAvailable.ObservedGeneration != opr.Object.GetGeneration() {
+		return ac
+	}
+
+	if operandAvailable.Status != "True" {
+		ac.Status = prometheusStatusToMSStatus(operandAvailable.Status)
+		if operandAvailable.Status == "Degraded" {
+			ac.Reason = fmt.Sprintf("%sDegraded", opr.name)
+		} else {
+			ac.Reason = fmt.Sprintf("%sNotAvailable", opr.name)
+		}
+		ac.Message = operandAvailable.Message
+		ac.LastTransitionTime = metav1.Now()
+		return ac
+	}
+	ac.Status = v1alpha1.ConditionTrue
+	ac.Reason = AvailableReason
+	ac.Message = AvailableMessage
+	ac.ObservedGeneration = generation
+	ac.LastTransitionTime = metav1.Now()
+	return ac
+}
+
+// updateReconciled updates "Reconciled" conditions based on the provided error value and
+// the operand "Reconciled" condition
+func updateReconciled(conditions []v1alpha1.Condition, opr Operand, generation int64, reconcileErr error) v1alpha1.Condition {
+	rc, cErr := getConditionByType(conditions, v1alpha1.ReconciledCondition)
+	if cErr != nil {
+		rc = v1alpha1.Condition{
+			Type:               v1alpha1.ReconciledCondition,
+			Status:             v1alpha1.ConditionUnknown,
+			Reason:             NoReason,
+			LastTransitionTime: metav1.Now(),
+		}
+	}
+	if reconcileErr != nil {
+		rc.Status = v1alpha1.ConditionFalse
+		rc.Message = reconcileErr.Error()
+		rc.Reason = FailedToReconcileReason
+		rc.LastTransitionTime = metav1.Now()
+		return rc
+	}
+	operandReconciled, reconcileErr := opr.getConditionByType(reconciled)
+
+	if reconcileErr != nil {
+		rc.Status = v1alpha1.ConditionUnknown
+		rc.Reason = fmt.Sprintf("%sNotReconciled", opr.name)
+		rc.Message = fmt.Sprintf("Cannot read %s status conditions", opr.name)
+		rc.LastTransitionTime = metav1.Now()
+		return rc
+	}
+
+	if operandReconciled.ObservedGeneration != opr.Object.GetGeneration() {
+		return rc
+	}
+
+	if operandReconciled.Status != "True" {
+		rc.Status = prometheusStatusToMSStatus(operandReconciled.Status)
+		rc.Reason = fmt.Sprintf("%sNotReconciled", opr.name)
+		rc.Message = operandReconciled.Message
+		rc.LastTransitionTime = metav1.Now()
+		return rc
+	}
+	rc.Status = v1alpha1.ConditionTrue
+	rc.Reason = ReconciledReason
+	rc.Message = SuccessfullyReconciledMessage
+	rc.ObservedGeneration = generation
+	rc.LastTransitionTime = metav1.Now()
+	return rc
+}
+
+func getConditionByType(conditions []v1alpha1.Condition, t v1alpha1.ConditionType) (v1alpha1.Condition, error) {
+	for _, c := range conditions {
+		if c.Type == t {
+			return c, nil
+		}
+	}
+	return v1alpha1.Condition{}, fmt.Errorf("ERROR: condition type %v not found", t)
+}
+
+func getConditionsFromObject(o client.Object) ([]v1alpha1.Condition, error) {
```

Comment:
Perhaps we could avoid this function by using some interface like:
```
type StatusReporting interface {
	Conditions() []Condition
	Geeneration() int64
}
```
And require types like `MonitoringStack` to implement this interface.

Reply:
@jan--f I was updating this PR this morning a bit (locally so far) and I think this ^^ can probably make few things more simple here. 
What do you think about this idea? The `Operand` type introduced here can then use this `StatusReporting` type instead of `client.Object` (but it's just theory. I haven't tried yet)

Reply:
Hm actually the theory was not correct :) I guess it won't help much with the conditions reading from the operands.